.github/workflows/release-candidate.yml

```diff
@@ -0,0 +1,107 @@
+name: Release Candidate
+
+on:
+  # Manual Trigger GH CLI -> gh workflow run release-candidate.yml --ref <branch-name>
+  workflow_dispatch:
+
+jobs:
+  run-python-tests:
+    uses: ./.github/workflows/python-versions.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-javascript-tests:
+    uses: ./.github/workflows/js-tests.yml
+    with:
+      ref: ${{ github.ref_name }}
+
+  run-py-prod-deps-smoke-test:
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  run-cypress-tests:
+    uses: ./.github/workflows/cypress.yml
+    with:
+      ref: ${{ github.ref_name }}
+      
+  build-release-candidate:
+    runs-on: ubuntu-latest
+
+    needs: [run-python-tests, run-javascript-tests, run-py-prod-deps-smoke-test, run-cypress-tests]
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Set Environment Variables
+        uses: ./.github/actions/branch
+        with:
+          branch: ${{ github.ref_name }}
+      - name: Set desired version from branch name
+        env:
+          BRANCH: ${{ env.BRANCH }}
+        run: |
+          echo "DESIRED_VERSION=$(echo "${BRANCH}" | tr -d "release/")" >> $GITHUB_ENV
+      - name: Set final versions for pre-release and update version
+        env: 
+          DESIRED_VERSION: ${{ env.DESIRED_VERSION }}
+        run: |
+          # Start at the latest existing RC for this version if one exists,
+          # so the following update will increment to the right RC version
+          LATEST_MATCHING_RC=$(curl https://pypi.org/pypi/streamlit/json | jq ".releases | keys | map(select(contains(\"rc\"))) | map(select(contains(\"$DESIRED_VERSION\"))) | sort | .[-1]" | tr --delete '"')
```

Comment:
This line is probably complex enough to warrant being written as a python script (especially since we're already calling a few scripts that live in the `scripts/` directory here) for readability's sake.

I'd be pretty happy with using a bunch of CLI-based tools to do this, but I'd imagine that figuring out what's going on here could be pretty rough vs reading something written to be equivalent in Python

The simpler string manipulations that we're doing around here should probably stay as-is (aside from only using one tool), though, since adding new scripts for those feels like overkill to me.

Reply:
Consolidated to using `sed` and added a python script for the `LATEST_MATCHING_RC` 👍🏼